### PR TITLE
Update spring boot version and azure spring boot version

### PR DESCRIPTION
**Pull request by moarychan**
Changelog:
- Update spring boot version to 2.4.5.
- Update azure spring boot version to 3.5.0
- Update maven-checkstyle-plugin version to 3.1.2.

### CODE REVIEW

**Comment by moarychan** [comment]
Hi @selvasingh, I am updating this doc [Spring with Cosmos on App Service](https://docs.microsoft.com/azure/developer/java/spring-framework/configure-spring-app-with-cosmos-db-on-app-service-linux), I saw that the last one commit was you, cloud you please review this  PR? or please help add @saragluna,  @backwind1233 as a code reviewer. Many thanks.

File: pom.xml [hunk 1]
```diff
@@ -1,95 +1,98 @@
...
+        <maven.build.timestamp.format>yyMMddHHmmssSSS</maven.build.timestamp.format>
+        <java.version>1.8</java.version>
+    </properties>
+
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>com.azure.spring</groupId>
+                <artifactId>azure-spring-boot-bom</artifactId>
+                <version>3.5.0</version>
```
Comment (stliu) [inline pom.xml:34]: define this in properties?

**Comment by stliu** [comment]
For this PR, I suggest only make change required, do not change the whole file or it's hard to tell what's changed

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**